Add role-based access for /secret command

uberdragon opened the pull request:
- Allow users with 'Beyond20 Admin' role to use /secret command
- Server owners retain full access (no breaking changes)
- Added GUILD_MEMBERS intent to bot.js for role checking
- Updated README with documentation for new role-based access feature

This change allows server owners to delegate /secret command access to trusted users by creating a 'Beyond20 Admin' role on their server, reducing the need for server owner intervention when managing channels.

uberdragon committed:
Add role-based access for /secret command

- Allow users with 'Beyond20 Admin' role to use /secret command
- Server owners retain full access (no breaking changes)
- Added GUILD_MEMBERS intent to bot.js for role checking
- Updated README with documentation for new role-based access feature

This change allows server owners to delegate /secret command access
to trusted users by creating a 'Beyond20 Admin' role on their server,
reducing the need for server owner intervention when managing channels.

kakaroto commented on commands/secret.js:
```diff
@@ -22,8 +22,11 @@ module.exports = {
-        if (interaction.user.id !== owner) {
-            return interaction.reply({content: 'Only the server owner can use this command.', ephemeral: true});
+        const isOwner = interaction.user.id === owner;
+        const hasAdminRole = interaction.member.roles.cache.some(role => role.name === 'Beyond20 Admin');
```
Double checking that the `cache` is always populated and we don't need to do anything about it, I found this: https://stackoverflow.com/questions/70795003/discordjs-guildmemberrolemanager-doesnt-have-a-cache
So I think you'd need to add a `interaction.inCachedGuild()` in case the interaction is not in a guild (i.e: someone sending `/secret` in a DM).

uberdragon replied:
Got it.  I originally didn't think about blocking DM since the command needs a channel as well.  I've updated the code to address this concern as well as the caching concern.  Booted up a dev bot and fully tested.  All appears to be working as expected.

kakaroto reviewed (changes requested):
Thank you for the contribution. I think that's a good idea, and while reviewing the code, I noticed one potential issue that needs fixing/testing.

On top of the `inCachedGuild()` (or maybe `inGuild()` ?)  issue mentioned below, I'm also not sure if we'd need to do `await interaction.member.fetch()` before accessing the cache for it to work reliably for everyone. I haven't been able to find any conclusive documentation on that.

uberdragon committed:
Address maintainer feedback for role-based access

- Add interaction.inGuild() check to prevent DM usage
- Add await interaction.member.fetch() for reliable role cache access
- Improve error handling and code reliability

Addresses concerns about:
- Guild context validation (prevents /secret in DMs)
- Reliable role checking with populated cache

kakaroto reviewed (approved):
Thank you for the quick fixes!

kakaroto commented:
Hi, giving you an update on this:
I tried to deploy it a couple of days ago but it kept failing due to the new intent, which is a privileged intent. When I went to enable it in the bot settings, it says that since the bot has more than 100 users, I'd need to get verified and then fill out a form to explain why the intent is necessary and what it will be used for, provide thorough explanations and a bunch of stuff (see https://support-dev.discord.com/hc/en-us/articles/6205754771351-How-do-I-get-Privileged-Intents-for-my-bot).

So... I need to better understand exactly why we need that intent in order to defend the case with the Discord crew. Looking into the intent GUILD_MEMBERS, it looks like it's only used for getting notified of members being added/removed from a server or added to a thread. I don't see anywhere anything that says it's required to get a member's roles list when they interact with the bot. 
Even found this which explains privileged intents a bit more than the official documentation and it implies that we should not need it, and that Discord would not favorably review my request for enabling the intent: https://gist.github.com/advaith1/e69bcc1cdd6d0087322734451f15aa2f
(Refering to the PRESENCE intent, it says)
> The presence intent is also required for the initial guild member list (which includes all members for small guilds and online members + members without roles for larger guilds), but note that your bot will not be whitelisted for the presence intent if you just need it for member caching. You should be fetching members/users when needed instead of relying on your bot's member/user cache.

Consider how it also says we should not need the GUILD_MEMBERS intent for getting a specific guild member by their name/ID, and how it talks about caching the guild member list, and fetching members when needed, I'm going to assume you only needed to add that intent in your original code submission because it wasn't caching the member roles, and with the intent, it had the roles cached cause it received the full members list, but with the addition of the `const member = await interaction.member.fetch();` in the last commit, the intent might not be necessary anymore.

I have very little available time right now, but once I do, I'll set up a dev environment again in order to test this and confirm whether or not the intent is unnecessary with the `member.fetch()`. If you are able to test that on your side and confirm it's not needed, that'd be a big help.

I'll keep you updated.
Thanks!
